clippy_lints/src/needless_bool.rs

```diff
@@ -204,6 +204,66 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for BoolComparison {
                 },
                 _ => (),
             }
+        } else if let ExprKind::Binary(
+            Spanned {
+                node: BinOpKind::Ne, ..
```

Comment:
since the only difference is the type of operation and some error messages, can you try to deduplicate this code with the above code? Mayne by moving it out into a function

Reply:
I was thinking, considering the difference in suggesting `!hint` or `hint`, on passing a closure (`Fn(Sugg) -> Sugg`) such as `|hint| !hint`.

Do you think there is a more idiomatic way of doing this?

Reply:
> Do you think there is a more idiomatic way of doing this?

Nope, that sounds good to me if that solves all the duplication.